code/game/objects/structures/traps.dm

```diff
@@ -63,7 +63,10 @@
 		return
 	if(ismob(AM))
 		var/mob/M = AM
-		if(M.mind in immune_minds || M.anti_magic_check())
+		if(M.mind in immune_minds)
+			return
+		if(M.anti_magic_check())
+			flare()
```

Comment:
?

Reply:
I found that while checking the code, this makes it so traps spend a charge when activating on a magic immune target